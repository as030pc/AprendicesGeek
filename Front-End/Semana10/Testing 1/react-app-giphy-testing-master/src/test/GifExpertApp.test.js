import React from 'react';
import {GifExpertApp} from '../GifExpertApp';
import {shallow} from 'enzyme'



describe('pruebas en gifexpert', ( ) =>  {
    test ('Debe mostrarse Correctamente', () => {
        const wrapp = shallow(<GifExpertApp/>)  //Renderizar el componente
        expect(wrapp).toMatchSnapshot();
    })
    test ('Debe mostrar una lista de categorias', () => {
        const categories = ["marvel", "Dragon Ball Z"]
        const wrapp = shallow(<GifExpertApp defaultCategories = {categories}/>)
        expect(wrapp).toMatchSnapshot();
        expect(wrapp.find('GifGrid').lenght).toBe(categories.lenght)  //toBe es para comparar es como ==
    })
})