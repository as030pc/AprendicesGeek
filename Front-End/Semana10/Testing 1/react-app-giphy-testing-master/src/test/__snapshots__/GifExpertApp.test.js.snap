// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en gifexpert Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="marvel"
      key="marvel"
    />
    <GifGrid
      category="Dragon Ball Z"
      key="Dragon Ball Z"
    />
  </ol>
</Fragment>
`;

exports[`pruebas en gifexpert Debe mostrarse Correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
